tests/eval_files/async912.py

```diff
@@ -0,0 +1,157 @@
+# ASYNCIO_NO_ERROR
+# ARG --enable=ASYNC100,ASYNC912
+# asyncio is tested in async912_asyncio. Cancelscopes in anyio are named the same
+# as in trio, so they're also tested with this file.
+
+# ASYNC100 has autofixes, but ASYNC912 does not. This leaves us with the option
+# of not testing both in the same file, or running with NOAUTOFIX.
+# NOAUTOFIX
+
+import trio
+
+
+def bar() -> bool:
+    return False
+
+
+async def foo():
+    # trivial cases where there is absolutely no `await` only triggers ASYNC100
+    with trio.move_on_after(0.1):  # ASYNC100: 9, "trio", "move_on_after"
+        ...
+    with trio.move_on_at(0.1):  # ASYNC100: 9, "trio", "move_on_at"
+        ...
+    with trio.fail_after(0.1):  # ASYNC100: 9, "trio", "fail_after"
+        ...
+    with trio.fail_at(0.1):  # ASYNC100: 9, "trio", "fail_at"
+        ...
+    with trio.CancelScope(0.1):  # ASYNC100: 9, "trio", "CancelScope"
+        ...
+
+    with trio.move_on_after(0.1):  # ASYNC912: 9
+        if bar():
+            await trio.lowlevel.checkpoint()
+    with trio.move_on_at(0.1):  # ASYNC912: 9
+        while bar():
+            await trio.lowlevel.checkpoint()
+    with trio.fail_after(0.1):  # ASYNC912: 9
+        try:
+            await trio.lowlevel.checkpoint()
+        except:
+            ...
+    with trio.fail_at(0.1):  # ASYNC912: 9
+        if bar():
+            await trio.lowlevel.checkpoint()
+    with trio.CancelScope(0.1):  # ASYNC912: 9
+        if bar():
+            await trio.lowlevel.checkpoint()
+    # ASYNC912 generally shares the same logic as other 91x codes, check respective
+    # eval files for more comprehensive tests.
+
+    # check we don't trigger on all context managers
+    with open(""):
+        ...
+
+    with trio.move_on_after(0.1):
+        await trio.lowlevel.checkpoint()
+
+    with trio.move_on_after(0.1):  # ASYNC912: 9
+        with trio.move_on_after(0.1):  # ASYNC912: 13
+            if bar():
+                await trio.lowlevel.checkpoint()
+
+    # We don't know which cancelscope will trigger first, so to avoid false
+    # positives on tricky-but-valid cases we don't raise any error for the outer one.
```

Comment:
nit, b/c clear in this case: I prefer "false alarm" vs "missed alarm", which is immediately descriptive.

By comparison 'false positive' vs 'false negative' is easy to confuse (is detecting a problem positive?), although still better than the abuse 'Type {1,2,I,II} error' terminology which gives absolutely no hints to the listener.

Reply:
Haha, I've seen you point this out about a dozen times and keep misremembering/forgetting which way to do it :D

Fixing